packages/genesys-spark-components/src/components/stable/gux-table/gux-all-row-select/gux-all-row-select.tsx

```diff
@@ -49,6 +50,26 @@ export class GuxAllRowSelect {
     this.inputElement.indeterminate = indeterminate;
   }
 
+  @Method()
+  async setCheckedState(checkedState: GuxAllRowSelectState): Promise<void> {
```

Comment:
If this function is useful we should be able to use it in the `updateSelectAllBoxState` method of `gux-table` and it should simplify the code/make it easier to read.